http-client-jdk/src/main/java/io/micronaut/http/client/jdk/JdkHttpClient.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.client.jdk;
+
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.io.ResourceResolver;
+import io.micronaut.core.type.Argument;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.bind.DefaultRequestBinderRegistry;
+import io.micronaut.http.bind.RequestBinderRegistry;
+import io.micronaut.http.client.BlockingHttpClient;
+import io.micronaut.http.client.DefaultHttpClientConfiguration;
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.HttpClientConfiguration;
+import io.micronaut.http.client.HttpVersionSelection;
+import io.micronaut.http.client.LoadBalancer;
+import io.micronaut.http.client.exceptions.HttpClientExceptionUtils;
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import io.micronaut.http.codec.MediaTypeCodecRegistry;
+import io.micronaut.json.JsonMapper;
+import io.micronaut.json.codec.JsonMediaTypeCodec;
+import io.micronaut.json.codec.JsonStreamMediaTypeCodec;
+import io.micronaut.runtime.ApplicationConfiguration;
+import org.reactivestreams.Publisher;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+
+/**
+ * {@link HttpClient} implementation for {@literal java.net.http.*} HTTP Client.
+ * @author Sergio del Amo
+ * @since 4.0.0
+ */
+@Internal
+@Experimental
+public class JdkHttpClient extends AbstractJdkHttpClient implements HttpClient {
```

Comment:
rename this to `DefaultJdkHttpClient` and add an interface called `JdkHttpClient` that would let me inject if I have both clients on the classpath

Reply:
I think this fixes it

https://github.com/micronaut-projects/micronaut-core/pull/8441/commits/69da8ad28ab427d143124d7a6c96fb1d5768752c

There's some nastiness around extracting cookies from the Netty client request, but I added tests to the test-suites (and pulled both client in to check JdkHttpClient injection)